### PR TITLE
Add DAO coin limit order bid-order query logic.

### PR DESCRIPTION
**Pull request by mattfoley8**

### CODE REVIEW

File: lib/db_utils.go [hunk 1]
```diff
@@ -6273,9 +6273,75 @@ func DBGetLowestDAOCoinAskOrders(txn *badger.Txn, inputOrder *DAOCoinLimitOrderE
...
+	//   * DAOCoinCreatorPKID
+	// Convert:
+	//   * OperationType from Ask to Bid
+	//   * PriceNanos to MAX_UINT256
+	//   * BlockHeight to MAX_UINT32
+	//   * Quantity to MAX_UINT256
+	queryOrder.OperationType = DAOCoinLimitOrderEntryOrderTypeBid
+	queryOrder.PriceNanos = *MaxUint256.Clone()
+	queryOrder.BlockHeight = math.MaxUint32
+	queryOrder.Quantity = *MaxUint256.Clone()
```
Comment (mattfoley8) [inline lib/db_utils.go:6298]: Not positive about this syntax.

File: lib/db_utils.go [hunk 2]
```diff
@@ -6273,9 +6273,75 @@ func DBGetLowestDAOCoinAskOrders(txn *badger.Txn, inputOrder *DAOCoinLimitOrderE
...
+	opts.Reverse = true // Go in reverse order since a higher bid price is better.
+	iterator := txn.NewIterator(opts)
+	defer iterator.Close()
+
+	// Seek first matching order.
+	orders := []*DAOCoinLimitOrderEntry{}
+
+	for iterator.Seek(key); iterator.ValidForPrefix(key) && requestedQuantity.GtUint64(0); iterator.Next() {
+		// If picking up from where you left off, skip the first order which has already been included.
+		if reflect.DeepEqual(key, startKey) {
```
Comment (lazynina) [inline lib/db_utils.go:6317]: reflection.DeepEqual is a mildly expensive operation. To speed this up, let's skip this check if:
1. startKey is nil
2. we've already seen the start key - add a boolean to track this.

File: lib/block_view_dao_coin_limit_order.go [hunk 3]
```diff
@@ -236,6 +246,9 @@ func (bav *UtxoView) _getNextLimitOrdersToFill(
 			if err != nil {
 				return nil, err
 			}
+		} else {
+			// TODO: switch to RuleErrorDAOCoinLimitOrderUnsupportedOperationType
+			return nil, fmt.Errorf("Invalid operation type")
```
Comment (lazynina) [inline lib/block_view_dao_coin_limit_order.go:251]:
```suggestion
			return nil, RuleErrorDAOCoinLimitOrderUnsupportedOperationType
```

**Review by lazynina — CHANGES_REQUESTED** [review]

File: lib/db_utils.go (hunk 1 above)
Comment (lazynina) [reply lib/db_utils.go:6298]: looks good to me!

File: lib/block_view_dao_coin_limit_order.go (hunk 3 above)
Comment (mattfoley8) [reply lib/block_view_dao_coin_limit_order.go:266]: This isn't defined yet. It's defined in https://github.com/deso-protocol/core/pull/253.

**Review by lazynina — APPROVED** [review]

File: lib/block_view_dao_coin_limit_order.go [hunk 4]
```diff
@@ -226,7 +226,17 @@ func (bav *UtxoView) _getNextLimitOrdersToFill(
...
+		if requestedOrder.OperationType == DAOCoinLimitOrderEntryOrderTypeAsk {
+			err := bav.Handle.View(func(txn *badger.Txn) error {
+				var err error
+				orders, err = DBGetHighestDAOCoinBidOrders(txn, requestedOrder, lastSeenKey)
+				return err
+			})
+
+			if err != nil {
+				return nil, err
+			}
```
[the remark this answers is not in the document]
Comment (lazynina) [reply lib/block_view_dao_coin_limit_order.go]: you right - this good

**Pull request merged**